src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -938,10 +958,21 @@ namespace winrt::Microsoft::Terminal::TerminalControl::implementation
 
     void TermControl::_KeyHandler(Input::KeyRoutedEventArgs const& e, const bool keyDown)
     {
+        if (e.OriginalKey() == VirtualKey::Enter)
+        {
+            return;
+        }
+
         // If the current focused element is a child element of searchbox,
         // we do not send this event up to terminal
-        if (_searchBox && _searchBox->ContainsFocus())
+        if (_searchBox && _searchBox->ContainsFocus() && _searchBox->TextBox().Text().empty())

```

Comment:
Please update the documentation of the condition as you have restricted it. 

Here as well I am not sure I understand why do we need the restriction. I mean now if the key event got here but the searchbox is focused are we sure it is OK to proceed? Cannot it result with sending a character to a terminal?

Optimally we don't need to check the `ContainsFocus` here at all. Instead we should make sure that SearchBoxControl handles the relevant key routed events.